File: trunk/theLink/example/cc/testclient.cc
/**
 *  \file       theLink/example/cc/testclient.cc
 *  \brief      \$Id$
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev$
 *  \author     EMail: aotto1968 at users.berlios.de
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */

#include  <iostream>
#include  <string>
#include  <cstring>
#include  <cstdlib>
#include  "ccmsgque.h"
#include  "debug.h"

using namespace std;
using namespace ccmsgque;

static string Get( MqC& ctx) {
  MqBufferC RET = ctx.ContextGetBuffer();
  ctx.SendSTART();
  ctx.SendEND_AND_WAIT("GTCX", MQ_TIMEOUT_USER);
  RET.Reset();
  RET.AppendC(ctx.ConfigGetName());
  RET.AppendC("+");
  RET.AppendC(ctx.ReadC());
  RET.AppendC(ctx.ReadC());
  RET.AppendC(ctx.ReadC());
  RET.AppendC(ctx.ReadC());
  RET.AppendC(ctx.ReadC());
  RET.AppendC(ctx.ReadC());
  return RET.GetC();
}

int MQ_CDECL main(int argc, MQ_CST argv[]) {

    // setup the clients
  MQ_CST t1[] =	  {argv[0], "--name", "c1", "--srvname", "s1"};
  vector<MQ_CST> LIST(t1, t1+5);

  if (argc > 1)
    LIST.insert(LIST.end(), argv+1, argv+argc);

  static MqC c0, c00, c01, c000, c1, c10, c100, c101;

  try {

    // create the "LINK"
#ifdef _WIN32
# define SRV  "../example/cc/testserver.exe"
#else
# define SRV  "../example/cc/testserver"
#endif

    c0.LinkCreateV(argv[0], "--name", "c0", "--debug", getenv("TS_DEBUG"), "@", SRV, "--name", "s0", NULL);
    c00.LinkCreateChildV(c0, argv[0], "--name", "c00", "--srvname", "s00", NULL);
    c01.LinkCreateChildV(c0, argv[0], "--name", "c01", "--srvname", "s01", NULL);
    c000.LinkCreateChildV(c00, argv[0], "--name", "c000", "--srvname", "s000", NULL);
    c1.LinkCreateVT(LIST);
    c10.LinkCreateChildV(c1, argv[0], "--name", "c10", "--srvname", "s10", NULL);
    c100.LinkCreateChildV(c10, argv[0], "--name", "c100", "--srvname", "s100", NULL);
    c101.LinkCreateChildV(c10, argv[0], "--name", "c101", "--srvname", "s101", NULL);

    // do the tests
    cout  <<  Get(c0)	<< endl
	  <<  Get(c00)	<< endl
	  <<  Get(c01)	<< endl
	  <<  Get(c000)	<< endl;

    cout  <<  Get(c1)	<< endl
	  <<  Get(c10)	<< endl
	  <<  Get(c100)	<< endl
	  <<  Get(c101)	<< endl;

  } catch (const exception& e) {
    c1.ErrorSet(e);
  }

  // do the cleanup
  c1.Exit();
}

File: tags/NHI1-0.16/theLink/example/go/testclient.go
/**
 *  \file       theLink/example/go/testclient.go
 *  \brief      \$Id$
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev$
 *  \author     EMail: aotto1968 at users.berlios.de
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */

package main

import (
  . "gomsgque"
    "os"
    "path"
    "fmt"
)

func Get(ctx *MqS) string {
  var RET string
  ctx.SendSTART()
  ctx.SendEND_AND_WAIT("GTCX", -1)
  RET += ctx.ConfigGetName()
  RET += "+"
  RET += ctx.ReadC()
  RET += ctx.ReadC()
  RET += ctx.ReadC()
  RET += ctx.ReadC()
  RET += ctx.ReadC()
  RET += ctx.ReadC()
  return RET
}

func main() {
  // create the object
  c0	:= NewMqS(nil,nil)
  c00	:= NewMqS(nil,nil)
  c01	:= NewMqS(nil,nil)
  c000	:= NewMqS(nil,nil)
  c1	:= NewMqS(nil,nil)
  c10	:= NewMqS(nil,nil)
  c100	:= NewMqS(nil,nil)
  c101	:= NewMqS(nil,nil)

  // catch errors and exit the app
  defer func() {
    if x := recover(); x != nil {
      c0.ErrorSet(x)
    }
    c0.Exit();
  }()

  // get server
  dir, _ := path.Split(os.Args[0])
  server := path.Join(dir, "testserver")
  LIST := append([]string{os.Args[0], "--name", "c1", "--srvname", "s1"}, os.Args[1:]...)

  // create the link
  c0.LinkCreate(os.Args[0], "--name", "c0", "--debug", os.Getenv("TS_DEBUG"), "@", server, "--silent", "--name", "s0")
  c00.LinkCreateChild(c0, os.Args[0], "--name", "c00", "--srvname", "s00")
  c01.LinkCreateChild(c0, os.Args[0], "--name", "c01", "--srvname", "s01")
  c000.LinkCreateChild(c00, os.Args[0], "--name", "c000", "--srvname", "s000")
  c1.LinkCreate(LIST...)
  c10.LinkCreateChild(c1, os.Args[0], "--name", "c10", "--srvname", "s10")
  c100.LinkCreateChild(c10, os.Args[0], "--name", "c100", "--srvname", "s100")
  c101.LinkCreateChild(c10, os.Args[0], "--name", "c101", "--srvname", "s101")

  // do the tests
  fmt.Fprintln(os.Stdout,Get(c0))
  fmt.Fprintln(os.Stdout,Get(c00))
  fmt.Fprintln(os.Stdout,Get(c01))
  fmt.Fprintln(os.Stdout,Get(c000))
  fmt.Fprintln(os.Stdout,Get(c1))
  fmt.Fprintln(os.Stdout,Get(c10))
  fmt.Fprintln(os.Stdout,Get(c100))
  fmt.Fprintln(os.Stdout,Get(c101))

  // cleanup
  c0.LinkDelete()
  c1.LinkDelete()
}

